tests/test_cli.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import collections
+import os
+import subprocess
+import yaml
+
+from conda_smithy import cli
+
+_thisdir = os.path.abspath(os.path.dirname(__file__))
+
+InitArgs = collections.namedtuple('ArgsObject', ('no_git_repo',
+                                                 'recipe_directory',
+                                                 'feedstock_directory',
+                                                 'variant_config_files'))
+
+RegenerateArgs = collections.namedtuple('ArgsObject', ('commit',
+                                                       'feedstock_directory',
+                                                       'variant_config_files'))
+
+
+def test_init(py_recipe):
+    """This is the command that takes the initial staged-recipe folder and turns it into a
+    feedstock"""
+    # actual parser doesn't matter.  It's used for initialization only
+    parser = argparse.ArgumentParser()
+    subparser = parser.add_subparsers()
+    init_obj = cli.Init(subparser)
+    recipe = py_recipe.recipe
+    # expected args object has
+    args = InitArgs(recipe_directory=os.path.join(recipe, 'recipe'),
+                    feedstock_directory=os.path.join(recipe, '{package.name}-feedstock'),
+                    no_git_repo=False,
+                    variant_config_files=os.path.join(recipe, 'config.yaml'))
+    init_obj(args)
+    destination = os.path.join(recipe, 'py-test-feedstock')
+    assert os.path.isdir(destination)
+
+
+def test_init_multiple_output_matrix(testing_workdir):
+    parser = argparse.ArgumentParser()
+    subparser = parser.add_subparsers()
+    init_obj = cli.Init(subparser)
+    recipe = os.path.join(_thisdir, 'recipes', 'multiple_outputs')
+    feedstock_dir = os.path.join(testing_workdir, 'multiple-outputs-test-feedstock')
+    args = InitArgs(recipe_directory=recipe,
+                    feedstock_directory=feedstock_dir,
+                    no_git_repo=False,
+                    variant_config_files=os.path.join(recipe, 'conda_build_config.yaml'))
+    init_obj(args)
+    matrix_dir = os.path.join(feedstock_dir, 'ci_support', 'matrix')
+    # the matrix should be consolidated among all outputs, as well as the top-level
+    # reqs. Only the top-level reqs should have indedependent config files,
+    # though - loops within outputs are contained in those top-level configs.
+    assert len(os.listdir(matrix_dir)) == 13
```

Comment:
Shouldn't this be 25? Tests failing because of this. (25 = 2 (libpq) * 2 (libpng) * 2 (jpeg) * 3 (os) + 1)

Reply:
Seems to be a problem in conda-build.  The matrix has two levels.  Things at the "top" level are things that affect the top-level recipe.  Essentially all of conda-forge recipes are just simply top-level recipes.  There's also output-level stuff.  In the recipe, take a look at where libpng, libpq, and jpeq occur.  libpng and libpq are top-level, but jpeg is output-level.  As an output-level thing, it should not enter into the outer loop, but instead should be a loop inside the build (a list in the generated conda_build_config.yaml), and ultimately would be a loop within a single build job on the given CI worker.

I hope that helps clarify things.  The test is correct, but the conda-build behavior is wrong.

Reply:
Thanks for the explanation. Makes sense to me.